### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by jpedryc**
#20 
Added initial `ReliqArts\GuidedImage\Demand\Image` & `ReliqArts\GuidedImage\Demand\ExistingImage` unit tests

### CODE REVIEW

File: tests/Unit/Demand/ImageTest.php [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    /**
+     * @dataProvider widthAndHeightDataProvider
+     * @covers ::__construct
+     * @covers ::getWidth
+     * @covers ::isValueConsideredNull
+     *
+     * @param mixed    $width
+     * @param null|int $expectedResult
+     */
+    public function testGetWidth($width, ?int $expectedResult)
```
Comment (reliq) [inline tests/Unit/Demand/ImageTest.php]: Please add return types; e.g. `: null`

**Review by reliq — CHANGES_REQUESTED** [review]
Thanks for this.

File: tests/Unit/Demand/ImageTest.php (hunk 1 above)
Comment (jpedryc) [reply tests/Unit/Demand/ImageTest.php]: Added 👍

File: composer.json [hunk 2]
```diff
@@ -43,7 +43,8 @@
     }
   },
   "scripts": {
-    "test": "vendor/bin/phpunit"
+    "test": "vendor/bin/phpunit",
+    "test-demands": "vendor/bin/phpunit --testdox tests/Unit/Demand"
```
Comment (reliq) [inline composer.json:47]: I don't think we'll need this going forward so I might remove it soon.

**Review by reliq — APPROVED** [review]
Thank you.

**Comment by jpedryc** [comment]
Sry for taking your time, it's my first ever push targeting not my repo - have a question 😄 Could you give me a hint about "not fully covered"? I thought the requirement is to have "touched" every function of a given class (with the annotation of `@cover`). Is the strategy of unit testing to cover also the methods of the parent classes?

**Comment by reliq** [comment]
No worries @jpedryc, you are correct. 

It was my mistake, I actually forgot I had already covered `.../Demand/Dummy.php`, `.../Demand/Resize.php` and `../Demand/Thumbnail.php`

I have gone ahead and merged this in. 

Thank you again.

Cheers! 🍻

**Comment by jpedryc** [comment]
Awesome 👍 Like the package, was looking for something like that couple months ago. Will try to help more here.

**Comment by reliq** [comment]
@jpedryc thank you!

**Pull request merged**